src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -1024,3 +1073,480 @@ def _read_semi_structured_file(self, path: str, format: str) -> DataFrame:
+                struct_schema = custom_schema
+            else:
+                logger.error(f"Invalid schema type: {type(custom_schema)}")
+                raise TypeError(f"Invalid schema type: {type(custom_schema)}.")
+
+        select_query = generate_select_query(
+            table, struct_schema, dbms_type, driver_info
+        )
+        logger.debug(f"Generated select query: {select_query}")
+        if column is None:
+            if (
+                lower_bound is not None
+                or upper_bound is not None
+                or num_partitions is not None
+            ):
+                logger.error("column is None but bounds or partitions are specified")
+                raise ValueError(
+                    "when column is not specified, lower_bound, upper_bound, num_partitions are expected to be None"
+                )
+            if predicates is None:
+                partitioned_queries = [select_query]
+            else:
+                partitioned_queries = generate_sql_with_predicates(
+                    select_query, predicates
+                )
+        else:
+            if lower_bound is None or upper_bound is None or num_partitions is None:
+                logger.error("column is specified but bounds or partitions are missing")
+                raise ValueError(
+                    "when column is specified, lower_bound, upper_bound, num_partitions must be specified"
+                )
+
+            column_type = None
+            for field in struct_schema.fields:
+                if field.name.lower() == column.lower():
+                    column_type = field.datatype
+            if column_type is None:
+                logger.error("Specified column does not exist in schema")
+                raise ValueError("Column does not exist")
+
+            if not isinstance(column_type, _NumericType) and not isinstance(
+                column_type, DateType
+            ):
+                logger.error(f"Unsupported column type: {column_type}")
+                raise ValueError(
+                    f"unsupported type {column_type}, we support numeric type and date type"
+                )
+            partitioned_queries = self._generate_partition(
+                select_query,
+                column_type,
+                column,
+                lower_bound,
+                upper_bound,
+                num_partitions,
+            )
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            # create temp table
+            snowflake_table_type = "temporary"
+            snowflake_table_name = random_name_for_temp_object(TempObjectType.TABLE)
+            create_table_sql = (
+                "CREATE "
+                f"{get_temp_type_for_object(self._session._use_scoped_temp_objects, True) if snowflake_table_type.lower() in TEMPORARY_STRING_SET else snowflake_table_type} "
+                "TABLE "
+                f"{snowflake_table_name} "
```

Comment:
May use parameter binding. Use `identifier(?)`.

Reply:
this cannot be done due to a bug, I posted it here: https://snowflake.slack.com/archives/C02KH7E6SHX/p1740702402032969

Reply:
updated